swarm/api/http/middleware.go

```diff
@@ -86,6 +87,47 @@ func InitLoggingResponseWriter(h http.Handler) http.Handler {
 	})
 }
 
+func InitUploadTag(h http.Handler, a *api.API) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var (
+			tagName        string
+			err            error
+			estimatedTotal = 0
+			contentType    = r.Header.Get("Content-Type")
+			contentLength  = r.Header.Get("Content-Length")
+			headerTag      = r.Header.Get(SwarmTagHeaderName)
+		)
+		if headerTag != "" {
+			tagName = headerTag
+			log.Trace("got tag name from http header", "tagName", tagName)
+		} else {
+			tagName = fmt.Sprintf("unnamed_tag_%d", time.Now().Unix())
+		}
+
+		log.Trace("trying to estimate tag size", "contentType", contentType, "contentLength", contentLength, "cl", r.ContentLength)
```

Comment:
why do you estimate when we can just calculate it?

Reply:
i did not find out where the actual chunk structure resides. a chunk is 4096 bytes, out of which the span is `x` bytes of size which i don't know. it's worth mentioning that our code only exposes chunk address and data, but the struct definition does not reflect the actual data structure!